.github/workflows/python-app.yml

```diff
@@ -54,21 +52,38 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 pytest tox cairo-lang
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        pip install flake8
     - name: Lint with flake8
       run: |
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with tox and pytest
+
+  coverage:
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    needs: [version, linter]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest tox
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Run tests and collect coverage
       run: |
-        tox
+        tox -e coverage -- --xml
+    - name: Upload coverage reports to Codecov with GitHub Action
+      uses: codecov/codecov-action@v3
 
   dist:
     runs-on: ubuntu-latest
-    needs: [test]
+    needs: [coverage]
```

Comment:
Keep in mind that this doesn't mean coverage must be high enough for `dist` to run. It only means that it will wait for the `coverage` job to finish running, i.e. to finish _sending_ the coverage report to Codecov. The results aren't delivered as part of the `codecov-action`, they are added to the pull request as a check.

Reply:
This check is added as part of the `.codecov.yml` file that is also part of this PR.

Reply:
Got it now. I don't think it is necessary to prevent dist from running if coverage is under 95 (keeping in mind the "test and release" action will succeed, but the codecov action won't). Interested in @martriay thoughts about this.

Reply:
i agree, based on our process (step 2) dist should only run on main or a release branch where coverage is assumed sufficient since every PR will perform the check.

so if coverage has no effect on whether dist runs or not, i don't see the reason to make it dependent

Reply:
i think we should remove the dependency